controller/instance_manager_controller.go

```diff
@@ -761,6 +889,19 @@ func (imc *InstanceManagerController) enqueueKubernetesNode(obj interface{}) {
 	}
 }
 
+func (imc *InstanceManagerController) enqueueSettingChange(obj interface{}) {
+	ims, err := imc.ds.ListInstanceManagers()
+	if err != nil {
+		utilruntime.HandleError(fmt.Errorf("failed to list instance manager on current node %v: %v",
+			imc.controllerID, err))
+		return
+	}
+
+	for _, im := range ims {
+		imc.enqueueInstanceManager(im)
```

Comment:
I might be wrong, but I think we should enqueue the IM. All IMs will catch the change when the setting gets updated. If we enqueue the node then all IMs will get enqueued + all the IMs on the IM node get enqueued again.

Reply:
You are not wrong, but each manager has their IM controller and all of them listen settings change now, so each of them should just enqueue their own instance managers, otherwise, there will be duplicated events to the queue?

Reply:
I see, you are right.

Reply:
Fixed.